### PR TITLE
Added profile header LEP resolver

### PR DESCRIPTION
**Pull request by pkev777**
* Added optional profile header lep resolver
* Use OptionalProfileHeaderLepResolver with UaaAuthentication and ProcessJwtAccessTokenDetails

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/security/DomainJwtAccessTokenDetailsPostProcessor.java [hunk 1]
```diff
@@ -6,14 +6,16 @@
...
+import com.icthh.xm.uaa.lep.keyresolver.OptionalProfileHeaderResolver;
 import org.springframework.security.oauth2.provider.OAuth2Authentication;
 import org.springframework.stereotype.Component;
 
 @Component
 @LepService(group = "security")
 public class DomainJwtAccessTokenDetailsPostProcessor {
 
-    @LogicExtensionPoint("ProcessJwtAccessTokenDetails")
+    @LogicExtensionPoint(value = "ProcessJwtAccessTokenDetails", resolver = OptionalProfileHeaderResolver.class)
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/security/DomainJwtAccessTokenDetailsPostProcessor.java:18]: WARNING!!! KEEP backward compatibility!!!

Comment (pkev777) [reply src/main/java/com/icthh/xm/uaa/security/DomainJwtAccessTokenDetailsPostProcessor.java:18]: Backward compatibility is present. New test was added to check lep function with profile header. And we already have existing test to check lep function without profile header

File: src/main/java/com/icthh/xm/uaa/lep/keyresolver/OptionalProfileHeaderResolver.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                                         LepManagerService managerService) {
+        return getHeaderValue("profile", String.class)
+            .map(SeparatorSegmentedLepKeyResolver::translateToLepConvention)
+            .map(translated -> new String[]{translated})
+            .orElseGet(() -> new String[0]);
+    }
+
+    private <T> Optional<T> getHeaderValue(final String headerName, final Class<T> valueType) {
+        return ofNullable(RequestContextHolder.getRequestAttributes())
+            .map(ServletRequestAttributes.class::cast)
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/lep/keyresolver/OptionalProfileHeaderResolver.java:31]: Are you sure that here we can cast to `ServletRequestAttributes` without an instanceof check?

Comment (pkev777) [reply src/main/java/com/icthh/xm/uaa/lep/keyresolver/OptionalProfileHeaderResolver.java:31]: hm, as I see yes. For web-based requests - implementation of requests attributes will be ServletRequestAttributes. And the method name: getHeaderValue - says that we expects only http request.

**Pull request merged**